nzedb/NameFixer.php

```diff
@@ -374,32 +376,105 @@ public function fixNamesWithPar2($time, $echo, $cats, $nameStatus, $show, $nntp)
 	}
 
 	/**
+	 * Attempts to fix release names using the mediainfo xml Unique_ID.
+	 *
+	 * @param int     $time 1: 24 hours, 2: no time limit
+	 * @param boolean $echo 1: change the name, anything else: preview of what could have been changed.
+	 * @param int     $cats 1: other categories, 2: all categories
+	 * @param         $nameStatus
+	 * @param         $show
+	 * @param string  $guidChar
+	 */
+	public function fixNamesWithMedia($time, $echo, $cats, $nameStatus, $show, $guidChar = '')
+	{
+		$type = 'UID, ';
+		$guid = ($guidChar === '') ? '' : ('AND rel.leftguid = '. $this->pdo->escapeString($guidChar));
+		if ($guid === '') {
+			$this->_echoStartMessage($time, 'mediainfo Unique_IDs');
+		}
+
+		// Only select releases we haven't checked here before
+		$preId = false;
+		if ($cats === 3) {
+			$query = sprintf('
+					SELECT rel.id AS releases_id, rel.name AS textstring
+					FROM releases rel
+					INNER JOIN release_unique ru ON (ru.releases_id = rel.id)
+					WHERE rel.nzbstatus = %d
+					AND rel.predb_id < 1',
+				NZB::NZB_ADDED
+			);
+			$cats = 2;
+			$preId = true;
+		} else {
+			$query = sprintf('
+					SELECT rel.id AS releases_id, rel.size AS relsize, rel.group_id, rel.categories_id, rel.name, rel.name AS textstring, rel.predb_id, rel.searchname, ru.releases_id, HEX(ru.uniqueid) AS uid
+					FROM releases rel
+					INNER JOIN release_unique ru ON (ru.releases_id = rel.id)
+					WHERE (rel.isrenamed = %d OR rel.categories_id IN (%d, %d))
+					AND rel.proc_uid = %d
+					%s',
+				self::IS_RENAMED_NONE,
+				Category::OTHER_MISC,
+				Category::OTHER_HASHED,
+				self::PROC_UID_NONE,
+				$guid
+			);
+		}
+
+		$releases = $this->_getReleases($time, $cats, $query);
+		if ($releases instanceof \Traversable && $releases !== false) {
+			$total = $releases->rowCount();
+			if ($total > 0) {
+				$this->_totalReleases = $total;
+				if ($guid === '') {
+					echo $this->pdo->log->primary(number_format($total) . ' unique ids to process.');
+				}
+				foreach ($releases as $rel) {
+					$this->checked++;
+					$this->done = $this->matched = false;
+					$this->checkName($rel, $echo, $type, $nameStatus, $show, $preId);
```

Comment:
Call uidCheck directly here.  We have probably already run checkName on this release before.